lib/x86_cpu/x86_llvm_env.ml

```diff
@@ -0,0 +1,76 @@
+open Bap.Std
+
+
+let base_var mode reg =
+  let module CPU = (val (X86_env.vars_of_mode mode)) in
+  let internal_reg = X86_asm_reg.decode reg in
+  match internal_reg with
+  | None -> None
+  | Some reg ->
+    if mode = X86_types.X86 then (
+      match reg with
```

Comment:
could we simplify this check by using polymorphic variants abbreviation patterns, e.g.,

```ocaml
match reg with
| #r64 | #r128 | #r256 | `RIP when mode = X86 -> 
   invalid_argf "register %s is not valid in the x86 mode" (string_of_reg reg) ()
| ...
```

Reply:
partially. the problem is some of the smaller variants are also infeasible (e.g., the lower part of r8)

Reply:
pushed a slightly improved version